lib/jekyll/site.rb

```diff
@@ -424,7 +424,14 @@ def configure_plugins
     private
     def configure_theme
       self.theme = nil
-      self.theme = Jekyll::Theme.new(config["theme"]) if config["theme"]
+      return unless config["theme"]
```

Comment:
That prints the warning if you don't set `theme` at all, which is not desired. i'll update it.